### ISSUE TITLE
NTOPNG crash every 3-4 hours ([LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil)

### ISSUE THREAD

**Issue by oronkessel1** [ntop/ntopng#4690]
Hi,

Since upgrading to v.4.2.201103 the service crash and I need to restart the service to bring the webpage up and start collecting data again.

Here is the log:

06/Nov/2020 10:57:21 [Ntop.cpp:2336] Setting local networks to 10.9.9.0/24
06/Nov/2020 10:57:21 [Redis.cpp:157] Successfully connected to redis 127.0.0.1:6379@0
06/Nov/2020 10:57:21 [Redis.cpp:157] Successfully connected to redis 127.0.0.1:6379@0
06/Nov/2020 10:57:21 [NtopPro.cpp:299] [LICENSE] Reading license from /etc/ntopng.license
06/Nov/2020 10:57:21 [NtopPro.cpp:429] [LICENSE] /etc/ntopng.license: found valid Enterprise L license
06/Nov/2020 10:57:21 [Ntop.cpp:2441] Registered interface tcp://127.0.0.1:1234 [id: 0]
06/Nov/2020 10:57:21 [main.cpp:312] PID stored in file /var/run/ntopng.pid
06/Nov/2020 10:57:21 [Geolocation.cpp:140] Loaded database dbip-asn-lite.mmdb [/usr/share/ntopng/httpdocs/geoip/dbip-asn-lite.mmdb][ip_version: 6]
06/Nov/2020 10:57:21 [Geolocation.cpp:140] Loaded database dbip-country-lite.mmdb [/usr/share/ntopng/httpdocs/geoip/dbip-country-lite.mmdb][ip_version: 6]
06/Nov/2020 10:57:21 [Geolocation.cpp:91] Using geolocation provided by DB-IP (https://db-ip.com)
06/Nov/2020 10:57:21 [HTTPserver.cpp:1529] Web server dirs [/usr/share/ntopng/httpdocs][/usr/share/ntopng/scripts]
06/Nov/2020 10:57:21 [HTTPserver.cpp:1532] HTTP server listening on 3000
06/Nov/2020 10:57:21 [Utils.cpp:764] User changed to ntopng
06/Nov/2020 10:57:21 [MySQLDB.cpp:824] Attempting to connect to MySQL for interface tcp://127.0.0.1:1234...
06/Nov/2020 10:57:21 [NetworkInterface.cpp:2627] Started flow dump loop on interface tcp://127.0.0.1:1234 [id: 0]...
06/Nov/2020 10:57:21 [MySQLDB.cpp:850] Successfully connected to MySQL [ntopng@localhost:3306] for interface tcp://127.0.0.1:1234
06/Nov/2020 10:57:21 [MySQLDB.cpp:824] Attempting to connect to MySQL for interface tcp://127.0.0.1:1234...
06/Nov/2020 10:57:21 [MySQLDB.cpp:850] Successfully connected to MySQL [ntopng@localhost:3306] for interface tcp://127.0.0.1:1234
06/Nov/2020 10:57:21 [MySQLDB.cpp:744] Disconnected from MySQL for interface tcp://127.0.0.1:1234...
06/Nov/2020 10:57:21 [MySQLDB.cpp:824] Attempting to connect to MySQL for interface tcp://127.0.0.1:1234...
06/Nov/2020 10:57:21 [MySQLDB.cpp:850] Successfully connected to MySQL [ntopng@localhost:3306] for interface tcp://127.0.0.1:1234
06/Nov/2020 10:57:21 [NetworkInterface.cpp:2593] Started flow user script hooks loop on interface tcp://127.0.0.1:1234 [id: 0]...
06/Nov/2020 10:57:21 [BatchedMySQLDB.cpp:69] Executing queryLoop()
06/Nov/2020 10:57:21 [MySQLDB.cpp:824] Attempting to connect to MySQL for interface tcp://127.0.0.1:1234...
06/Nov/2020 10:57:21 [main.cpp:382] Working directory: /home/lib/ntopng
06/Nov/2020 10:57:21 [main.cpp:384] Scripts/HTML pages directory: /usr/share/ntopng
06/Nov/2020 10:57:21 [MySQLDB.cpp:850] Successfully connected to MySQL [ntopng@localhost:3306] for interface tcp://127.0.0.1:1234
06/Nov/2020 10:57:21 [Ntop.cpp:440] Welcome to ntopng x86_64 v.4.2.201103 - (C) 1998-20 ntop.org
06/Nov/2020 10:57:21 [NtopPro.cpp:701] [LICENSE] System Id:	0F983802B205A206
06/Nov/2020 10:57:21 [NtopPro.cpp:702] [LICENSE] Edition:	Enterprise L
06/Nov/2020 10:57:21 [NtopPro.cpp:703] [LICENSE] License Type:	Permanent License
xxxxxxxxxxxx I removed xxxxxxxxxx
06/Nov/2020 10:57:21 [NtopPro.cpp:727] [LICENSE] Maintenance:	Until Mon May 10 09:28:45 2021 [184 days left]
06/Nov/2020 10:57:21 [PeriodicActivities.cpp:109] Started periodic activities loop...
06/Nov/2020 10:57:22 [startup.lua:50] Processing startup.lua: please hold on...
06/Nov/2020 10:57:22 [startup.lua:144] [lists_utils.lua:758] Refreshing category lists...
06/Nov/2020 10:57:22 [startup.lua:144] [lists_utils.lua:696] Category Lists (694 hosts, 4516 IPs, 98 JA3) loaded in 0 sec
06/Nov/2020 10:57:23 [startup.lua:218] Startup completed: ntopng is now operational
06/Nov/2020 10:57:23 [PeriodicActivities.cpp:172] Each periodic activity script will use 2 threads
06/Nov/2020 10:57:23 [NetworkInterface.cpp:2735] Started packet polling on interface tcp://127.0.0.1:1234 [id: 0]...
06/Nov/2020 10:57:24 [ZMQCollectorInterface.cpp:256] Collecting flows on tcp://127.0.0.1:1234
06/Nov/2020 11:00:00 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
06/Nov/2020 11:00:00 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
06/Nov/2020 11:00:00 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
06/Nov/2020 11:00:00 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
06/Nov/2020 11:00:00 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
06/Nov/2020 1

**Comment by oronkessel1** [comment]
I get these errors:

07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil
07/Nov/2020 03:10:02 [LuaEngineNtop.cpp.inc:37] ERROR: ntop_snmpv3_batch_get : expected string[@pos 4], got nil

**Comment by simonemainardi** [comment]
crash possibly fixed as part of https://github.com/ntop/ntopng/issues/4678. New builds are in progress. Please, hold on a couple of hours, update and report.

In parallel, my colleague @lucaderi is looking at the issue with SNMP.

**Comment by lucaderi** [comment]
@simonemainardi Not sure your fix will help as it is not a SNMPv3 issue
@oronkessel1 Do you have a IP we can use to reproduce the problem?

**Comment by oronkessel1** [comment]
@lucaderi I'm scanning my core router and it was working fine on older ver.
what do you need exactly ?

**Comment by simonemainardi** [comment]
@lucaderi as MySQL is in use also in this instance, I think that that - not SNMP - could have been the cause of the crash. See #4678

**Comment by oronkessel1** [comment]
is there a way to roll back to older ver ?

**Comment by simonemainardi** [comment]
do not rollback please. Just run the update as suggested and tell if you are seeing the crash.

**Comment by simonemainardi** [comment]
crash is reportedly solved: https://github.com/ntop/ntopng/issues/4678

**Comment by oronkessel1** [comment]
ok, upgraded.
I will let you know

**Comment by lucaderi** [comment]
Thank  you. If SNMPv3 is still not working please open a new ticket and specify the exact configuration you have used (except sensitive info such as username)
